### PR TITLE
Search modal

### PR DESCRIPTION
**Pull request by carlosgauci**
- Added tutorial search modal (placeholder search result)

### CODE REVIEW

File: themes/haystack/layouts/_default/tutorials.html [hunk 1]
```diff
@@ -194,10 +198,179 @@ <h2>{{ .Params.contribute.title }}</h2>
...
+
+        if (response.status === 200) {
+          const result = await response.json();
+
+          if (result.results[0].documents.length > 0) {
+            {{/*  build cards  */}}
+            result.results[0].documents.forEach((doc) => {
+              const card = document.createElement("a")
+              card.classList.add("search-modal-card")
+              card.href = `/tutorials/${doc.meta.file_name.toLowerCase().replace(".txt", "")}`;
```
Comment (bilgeyucel) [inline themes/haystack/layouts/_default/tutorials.html:273]: I updated the tutorial response. Now, every tutorial has `headlines` information in the `meta` field. Can we use this information to direct the users precisely to the place where the text is retrieved from? 

**Here's an example**
This is a headline of a result:
```
headlines: [
0: {level: 1, headline: "Initializing the PromptNode"}
1: {level: 1, start_idx: 455, headline: "Defining the Pipeline"}
]
```
This is the `meta.link` of the same result: `tutorials/22_pipeline_with_promptnode`
The link that this result item directs to should be: `tutorials/22_pipeline_with_promptnode#initializing-the-promptnode` 
Basically, you need to take only the first item of headlines and add it to the end of the link with `-` instead of space.

**Review by bilgeyucel — CHANGES_REQUESTED** [review]

**Review by bilgeyucel — APPROVED** [review]
